### PR TITLE
Return type should be 'long' (or any primitive)

### LINKED ISSUE #2543: Return type should be 'long' (or any primitive)
Return type should be 'long' (or any primitive) where Redis does not return 'nil'.

### PR DESCRIPTION
**Pull request by sazzad16**
Resolves #2543

### CODE REVIEW

**Review by dengliming — APPROVED** [review]

**Review by mina-asham — APPROVED** [review]
Due to the size of the review I was mostly sampling here, as far as I can see and notice this is fine :)

One comment though, this can target an earlier release than 4.0.0, this change is backward compatible (going from wrapper primitives to primitives is a backward compatible change in general, the other way is the one that might be incompatible in some cases)

**Comment by sazzad16** [comment]
> this can target an earlier release than 4.0.0, this change is backward compatible (going from wrapper primitives to primitives is a backward compatible change in general, the other way is the one that might be incompatible in some cases)

@mina-asham I'm okay with it. But others may not agree.

I chose 4.0.0 as a safer option so that existing `jedis.xlen("xlen-stream").longValue()` kind of codes don't get broken when updated.

@gkorland ping!

File: src/main/java/redis/clients/jedis/BinaryJedis.java [hunk 1]
```diff
@@ -4366,20 +4366,15 @@ public String migrate(final String host, final int port, final int destinationDB
     return client.getStatusCodeReply();
   }
 
-  /**
-   * Syncrhonous replication of Redis as described here: http://antirez.com/news/66 Since Java
-   * Object class has implemented "wait" method, we cannot use it, so I had to change the name of
```
[the remark this answers is not in the document]
Comment (sazzad16) [reply src/main/java/redis/clients/jedis/BinaryJedis.java:4372]: @gkorland The interface, BasicCommands, has better JDoc https://github.com/redis/jedis/pull/2547/files#diff-d54b47364c0fa44a622800fecc2a8edf5723746cf27feae7d136ba7357701d03R202-R216

File: src/test/java/redis/clients/jedis/tests/PipeliningTest.java [hunk 2]
```diff
@@ -303,7 +303,7 @@ public void multiWithSync() {
   @Test
   public void multiWatch() {
     final String key = "foo";
-    assertEquals(Long.valueOf(5L), jedis.incrBy(key, 5L));
+    assertEquals(5, jedis.incrBy(key, 5L));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/PipeliningTest.java:306]:
```suggestion
    assertEquals(5L, jedis.incrBy(key, 5L));
```

File: src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java [hunk 3]
```diff
@@ -83,23 +83,17 @@ public void exists() {
...
-    assertTrue(reply);
+    assertTrue(jedis.exists("foo"));
 
-    reply = jedis.exists(bfoo);
-    assertTrue(reply);
+    assertTrue(jedis.exists(bfoo));
 
-    long lreply = jedis.del("foo");
-    assertEquals(1, lreply);
+    assertEquals(1, jedis.del("foo"));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java:90]:
```suggestion
    assertEquals(1L, jedis.del("foo"));
```

File: src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java [hunk 4]
```diff
@@ -83,23 +83,17 @@ public void exists() {
...
 
-    lreply = jedis.del(bfoo);
-    assertEquals(1, lreply);
+    assertEquals(1, jedis.del(bfoo));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java:92]:
```suggestion
    assertEquals(1L, jedis.del(bfoo));
```

File: src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java [hunk 5]
```diff
@@ -110,14 +104,11 @@ public void existsMany() {
     status = jedis.set("foo2", "bar2");
     assertEquals("OK", status);
 
-    long reply = jedis.exists("foo1", "foo2");
-    assertEquals(2, reply);
+    assertEquals(2, jedis.exists("foo1", "foo2"));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java:107]:
```suggestion
    assertEquals(2L, jedis.exists("foo1", "foo2"));
```

File: src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java [hunk 6]
```diff
@@ -110,14 +104,11 @@ public void existsMany() {
...
 
-    long lreply = jedis.del("foo1");
-    assertEquals(1, lreply);
+    assertEquals(1, jedis.del("foo1"));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java:109]:
```suggestion
    assertEquals(1L, jedis.del("foo1"));
```

File: src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java [hunk 7]
```diff
@@ -110,14 +104,11 @@ public void existsMany() {
...
 
-    reply = jedis.exists("foo1", "foo2");
-    assertEquals(1, reply);
+    assertEquals(1, jedis.exists("foo1", "foo2"));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java:111]:
```suggestion
    assertEquals(1L, jedis.exists("foo1", "foo2"));
```

File: src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java [hunk 8]
```diff
@@ -126,46 +117,34 @@ public void del() {
     jedis.set("foo2", "bar2");
     jedis.set("foo3", "bar3");
 
-    long reply = jedis.del("foo1", "foo2", "foo3");
-    assertEquals(3, reply);
+    assertEquals(3, jedis.del("foo1", "foo2", "foo3"));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java:120]:
```suggestion
    assertEquals(3L, jedis.del("foo1", "foo2", "foo3"));
```

File: src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java [hunk 9]
```diff
@@ -126,46 +117,34 @@ public void del() {
...
-    assertFalse(breply);
+    assertFalse(jedis.exists("foo1"));
+    assertFalse(jedis.exists("foo2"));
+    assertFalse(jedis.exists("foo3"));
 
     jedis.set("foo1", "bar1");
 
-    reply = jedis.del("foo1", "foo2");
-    assertEquals(1, reply);
+    assertEquals(1, jedis.del("foo1", "foo2"));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java:128]:
```suggestion
    assertEquals(1L, jedis.del("foo1", "foo2"));
```

File: src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java [hunk 10]
```diff
@@ -126,46 +117,34 @@ public void del() {
...
 
-    reply = jedis.del("foo1", "foo2");
-    assertEquals(0, reply);
+    assertEquals(0, jedis.del("foo1", "foo2"));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java:130]:
```suggestion
    assertEquals(0L, jedis.del("foo1", "foo2"));
```

File: src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java [hunk 11]
```diff
@@ -126,46 +117,34 @@ public void del() {
...
 
     // Binary ...
     jedis.set(bfoo1, bbar1);
     jedis.set(bfoo2, bbar2);
     jedis.set(bfoo3, bbar3);
 
-    reply = jedis.del(bfoo1, bfoo2, bfoo3);
-    assertEquals(3, reply);
+    assertEquals(3, jedis.del(bfoo1, bfoo2, bfoo3));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/AllKindOfValuesCommandsTest.java:137]:
```suggestion
    assertEquals(3L, jedis.del(bfoo1, bfoo2, bfoo3));
```

**Review by gkorland — CHANGES_REQUESTED** [review]

File: src/test/java/redis/clients/jedis/tests/commands/BinaryValuesCommandsTest.java [hunk 12]
```diff
@@ -134,37 +124,31 @@ public void setAndKeepttl() {
...
-    assertTrue(Arrays.equals(binaryValue, value));
+    assertNull(jedis.getSet(bfoo, binaryValue));
+    assertArrayEquals(binaryValue, jedis.get(bfoo));
   }
 
   @Test
   public void getDel() {
-    String status = jedis.set(bfoo, bbar);
-    assertEquals("OK", status);
+    assertEquals("OK", jedis.set(bfoo, bbar));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/BinaryValuesCommandsTest.java:149]:
```suggestion
    assertEquals(Keyword.OK.name(), jedis.set(bfoo, bbar));
```

File: src/test/java/redis/clients/jedis/tests/commands/BinaryValuesCommandsTest.java [hunk 13]
```diff
@@ -191,8 +175,7 @@ public void getEx() {
     assertTrue(ttl > 30 && ttl <= 40);
 
     assertArrayEquals(bbar, jedis.getEx(bfoo, GetExParams.getExParams().persist()));
-    ttl = jedis.ttl(bfoo);
-    assertEquals(-1, ttl);
+    assertEquals(-1, jedis.ttl(bfoo));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/BinaryValuesCommandsTest.java:178]:
```suggestion
    assertEquals(-1L, jedis.ttl(bfoo));
```

**Review by gkorland — CHANGES_REQUESTED** [review]

File: src/test/java/redis/clients/jedis/tests/commands/StringValuesCommandsTest.java [hunk 14]
```diff
@@ -228,22 +211,21 @@ public void substr() {
...
+    jedis.set("s", str);
+    assertEquals(str.length(), jedis.strlen("s"));
   }
 
   @Test
   public void incrLargeNumbers() {
-    long value = jedis.incr("foo");
-    assertEquals(1, value);
+    assertEquals(1, jedis.incr("foo"));
     assertEquals(1L + Integer.MAX_VALUE, (long) jedis.incrBy("foo", Integer.MAX_VALUE));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/StringValuesCommandsTest.java:222]:
```suggestion
    assertEquals(1L + Integer.MAX_VALUE, jedis.incrBy("foo", Integer.MAX_VALUE));
```

File: src/test/java/redis/clients/jedis/tests/commands/StreamsCommandsTest.java [hunk 15]
```diff
@@ -139,9 +139,9 @@ public void xaddWithParams() {
 
     // minid
     jedis.xadd("xadd-stream3", map6, XAddParams.xAddParams().minId("2").id("2"));
-    assertEquals(1, jedis.xlen("xadd-stream3").longValue());
+    assertEquals(1, jedis.xlen("xadd-stream3"));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/StreamsCommandsTest.java:142]:
```suggestion
    assertEquals(1L, jedis.xlen("xadd-stream3"));
```

File: src/test/java/redis/clients/jedis/tests/commands/StreamsCommandsTest.java [hunk 16]
```diff
@@ -139,9 +139,9 @@ public void xaddWithParams() {
...
     jedis.xadd("xadd-stream3", map6, XAddParams.xAddParams().minId("4").id("3"));
-    assertEquals(0, jedis.xlen("xadd-stream3").longValue());
+    assertEquals(0, jedis.xlen("xadd-stream3"));
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/StreamsCommandsTest.java:144]:
```suggestion
    assertEquals(0L, jedis.xlen("xadd-stream3"));
```

**Review by gkorland — APPROVED** [review]

**Pull request merged**